webroot/src/pages/SelectPrivileges/SelectPrivileges.ts

```diff
@@ -0,0 +1,380 @@
+                date = moment(expireDate).format(displayDateFormat);
+            }
+        }
+
+        return date;
+    }
+
+    get subTotalList(): Array<number> {
+        return this.selectedStatePurchaseDataList?.map((purchaseInfo) => {
+            const militaryDiscount = purchaseInfo.isMilitaryDiscountActive && purchaseInfo.militaryDiscountAmount
+                ? purchaseInfo.militaryDiscountAmount : 0;
+
+            return ((purchaseInfo?.fee || 0) + (this.currentCompactCommissionFee || 0) - (militaryDiscount));
+        });
+    }
+
+    get selectPrivilegesTitleText(): string {
+        return this.$t('licensing.selectPrivileges');
+    }
+
+    get shouldShowJurisprudenceModal(): boolean {
+        return !!this.jurisprudencePendingConfirmation;
+    }
+
+    get areAllJurisprudenceConfirmed(): boolean {
+        let allConfirmed = true;
+
+        if (this.formData?.jurisprudenceConfirmations) {
+            const jurisprudenceConfirmations = Object.keys(this.formData.jurisprudenceConfirmations);
+
+            jurisprudenceConfirmations.forEach((state) => {
+                if (!this.formData.jurisprudenceConfirmations[state].value) {
+                    allConfirmed = false;
+                }
+            });
+        }
+
+        return allConfirmed;
+    }
+
+    //
+    // Methods
+    //
+    initFormInputs(): void {
+        const initFormData: any = {
+            stateCheckList: [],
+            jurisprudenceConfirmations: {},
+            submit: new FormInput({
+                isSubmitInput: true,
+                id: 'submit',
+            }),
+        };
+
+        this.purchaseDataList?.forEach((purchaseOption) => {
+            const { jurisdiction = new State() } = purchaseOption;
+
+            if (jurisdiction) {
+                const { abbrev } = jurisdiction;
+
+                if (abbrev) {
+                    // const shouldDisable = this.alreadyObtainedPrivilegeStates.includes(abbrev);
+
+                    initFormData.stateCheckList.push(new FormInput({
+                        id: abbrev,
+                        name: `${abbrev}-check`,
+                        label: jurisdiction.name(),
+                        value: false,
+                        isDisabled: true
+                    }));
+                }
+            }
+        });
+
+        this.formData = reactive(initFormData);
+    }
+
+    handleSubmit() {
+        // This function is just a stub as it is the boundary to the next feature to be developed
+        if (this.isAtLeastOnePrivilegeChosen && this.areAllJurisprudenceConfirmed) {
+            this.startFormLoading();
+            console.log(this.formData);
+            this.endFormLoading();
+        }
+    }
+
+    handleJurisprudenceClicked(privilegePurchaseInformation) {
+        const stateAbbrev = privilegePurchaseInformation.jurisdiction.abbrev;
+        const newValue = this.formData.jurisprudenceConfirmations[stateAbbrev].value;
+
+        if (newValue === true) {
```

Comment:
I assume `newValue` is always a boolean? I also might rename this to something more specific like `confirmationStatus` (or something else that mirrors your other naming conventions)

```suggestion
        if (newValue) {
```